python/feathub/processors/flink/table_builder/flink_table_builder.py

```diff
@@ -408,6 +407,30 @@ def _get_table_from_derived_feature_view(
             right_table = table_by_names[right_table_name]
             right_table_descriptor = self.registry.get_features(right_table_name)
             redis_source = get_redis_source(right_table_descriptor)
+            right_table = optimize_redis_source_lookup_map(
```

Comment:
Would it be more readable to call this method only when `redis_source is not None`?